src/main/rx/sbus_channels.c

```diff
@@ -68,12 +68,15 @@ uint8_t sbusChannelsDecode(rxRuntimeConfig_t *rxRuntimeConfig, const sbusChannel
         sbusChannelData[17] = SBUS_DIGITAL_CHANNEL_MIN;
     }
 
-    if (channels->flags & SBUS_FLAG_SIGNAL_LOSS) {
-    }
     if (channels->flags & SBUS_FLAG_FAILSAFE_ACTIVE) {
         // internal failsafe enabled and rx failsafe flag set
-        // RX *should* still be sending valid channel data, so use it.
-        return RX_FRAME_COMPLETE | RX_FRAME_FAILSAFE;
+        // RX *should* still be sending valid channel data (repeated), so use it.
+        return RX_FRAME_COMPLETE | RX_FRAME_FAILSAFE | RX_FRAME_DROPPED;
+    }
+
+    if (channels->flags & SBUS_FLAG_SIGNAL_LOSS) {
+    	// The received data is a repeat of the last valid data so can be considered complete.
+    	return RX_FRAME_COMPLETE | RX_FRAME_DROPPED;
```

Comment:
This should not set `RX_FRAME_COMPLETE`, in order to not interfere with the signal loss detection.

Reply:
@SteveCEvans: This is still an open point. The way I understand the code is that if `RX_FRAME_COMPLETE` is set here, if an RX uses `SBUS_FLAG_SIGNAL_LOSS` to indicate signal loss, no failsafe will be triggered, with potentially disastrous consequences.

Reply:
Update: I had a look at https://github.com/betaflight/betaflight/blob/master/src/main/rx/rx.c, and the way that `needRxSignalBefore` is handled means indeed that if we send `RX_FRAME_COMPLETE`, signal loss will never trigger a failsafe, which is a severe safety issue. Please change.